src/cargo/core/compiler/job_queue.rs

```diff
@@ -583,6 +583,17 @@ impl<'cfg> DrainState<'cfg> {
             }
         }
 
+        // If multiple pieces of work are waiting in the pending queue, we can sort it according to
+        // their priorities: higher priorities should be scheduled sooner.
+        if self.pending_queue.len() > 1 {
```

Comment:
I suppose that `sort_by` does a check like this anyway, so I'd remove it and then the diff of this PR could be even (2 lines) shorter! :laughing:

Reply:
[For sure](https://doc.rust-lang.org/src/alloc/slice.rs.html#1099) ^^

Reply:
I've pushed that change